Users api (#176)
metricly/metricly-docs: content/api/api-users.md

    "assumedReadOnlyIdentity": false,
    "readOnly": false
  }
}
```

{{% /expand %}}

---
## GET from /users/{id}/featureusage
{{< button theme="info" href="https://app.metricly.com/swagger-ui.html#!/users/getFeatureUsageUsingGET" >}} GET {{< /button >}} Use this endpoint to check the setup status and feature usage of a user account.

{{% expand "View method details."%}}

### Request URL

 `https://app.metricly.com/users/{id}/featureusage`

### CURL

The following example includes the users unique **id** in the Request URL.
```
curl -X GET --header 'Accept: application/json' 'https://app.metricly.com/users/76502/featureusage'
```

### Response Body

The following response body shows a breakdown of actions available in CloudWisdom that have been taken by the user. This user has created at least one policy, dashboard, datasource, and notification. This account has been 100% configured.
```
{
  "featureUsage": {
    "id": 76502,
    "featuresUsed": {
      "createdPolicy": true,
      "createdDatasource": true,
      "createdDashboard": true,
      "createdNotification": true
    },
    "configuredPercentage": 100
  }
}
```
{{% /expand %}}

---
## POST to /users/{id}/password

> {{< button theme="success" href="https://app.metricly.com/swagger-ui.html#!/users/updatePasswordUsingPOST" >}} POST {{< /button >}} Use this endpoint to update the password of a user.

Reviewer: ```suggestion
{{< button theme="success" href="https://app.metricly.com/swagger-ui.html#!/users/updatePasswordUsingPOST" >}} POST {{< /button >}} Use this endpoint to update the password for your user.
```

{{% expand "View method details."%}}

### Request URL

>  `https://app.metricly.com/users/{id}/password?oldPassword={OLDPASS}&newPassword={NEWPASS}`

Reviewer: ```suggestion
 `https://app.metricly.com/users/{id}/password?oldPassword={old-password}&newPassword={new-password}`
```

      "metricsViewSetting": "tree"
    },
    "tenantProperties": {
      "netuitiveAccountId": "123456776543",
      "tenantType": "CUSTOMER",
      "tenantStatus": "ACTIVE",
      "productIds": [
        "monitoring",
        "cost_aws"
      ],
      "trialExpiresAt": null,
      "name": "Metricly Support Testing",
      "datasourceTrialCountLimit": 2
    },
    "customElementTypeIcons": {},
    "roles": [
      "Administrator"
    ],
    "administrator": true,
    "globalAdministrator": false,
    "globalReadOnly": false,
    "assumedIdentity": false,
    "assumedReadOnlyIdentity": false,
    "readOnly": false
  }
}
```
{{% /expand %}}

---

## How to Preconfigure a New User

Admins can create new users through the User API by sending a POST call to **/users**. In addition to basic setup, admins can use **properties** to define config such as the user's landing page, theme, and even set them up with a favorite dashboard. Follow this walkthrough to learn how to preconfigure a user.

{{% expand "View Walkthrough." %}}

1\. Build a basic JSON template for creating a user. This is an end user, so set the user to **ReadOnly** `True`.

{{% notice tip %}}

> Do not send a basic template to create the user; you cannot edit the user's preference details once it has been created -- only the  and **roles** and **permissions** (administrator, globalAdministrator, readOnly, etc) settings. Finish setting up all properties first.

Reviewer: ```suggestion
Do not send a basic template to create the user; you cannot edit the user's preference details once it has been created -- only the **roles** (Administrator vs. Read Only) settings can be modified after initial creation. Finish setting up all properties first.
```
